#171186789 Deploy base application to Heroku

Moise1 opened the pull request:
#### What does this PR do?
- It sets up both staging and production Heroku environments (pipelines)  for the app's manageable deployment.

#### Description of Task to be completed?
- configure Procfile
- creating staging and production pipelines
- configure config.js

#### How should this be manually tested?
- run `git clone https://github.com/andela/knights-bn-backend.git` in your terminal
- run `cd knights-bn-backend` in your terminal
- run `git checkout  ch-heroku-app-deployment-170947654 `
- run `npm install` to install all packages. 
- copy  the following link in your Postman's  address bar: `https://knights-bn-backend-staging.herokuapp.com/api/v1/auth/signup`
- Copy and paste this URL in your browser: `https://knights-bn-backend-staging.herokuapp.com/` and you should expect to see a similar page to the second screenshot in the PR.

- run the following in your Postman's request body: 

```
{
"firstName": "Kabera",
"lastName" : "Kaneza",
"gender":  "Male",
"passportNumber": "88877712t",
"email" : "email@gmail.com",
"password": "muneza12345"
}
```
#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#171186789](https://www.pivotaltracker.com/n/projects/2431452)

#### Screenshots (if appropriate)
**Heroku pipelines** 
<img width="1440" alt="Screen Shot 2020-02-17 at 12 00 28" src="https://user-images.githubusercontent.com/25052255/74643311-30389d80-517d-11ea-9fae-aa4f30ca0a99.png">

**Postman - Heroku result** 
<img width="1135" alt="Screen Shot 2020-02-20 at 12 09 05" src="https://user-images.githubusercontent.com/25052255/74923869-3081a480-53da-11ea-9f8c-d3883f725eae.png">

Niyonsengaeric commented:
can you`squash your commits` to make it one commit. 
It should be better if you put all the steps required to reach your feature under `How should this be manually tested?` in your PR description.

EugeneMunya commented:
@Moise1 nice work, but remember to change swagger host url to heroku not locahost

Moise1 committed:
ch(Heroku): Deploy base application to Heroku

- configure Procfile
- creating staging and production pipelines
- configure config.js

[Start: #170947654]

Moise1 commented:
@EugeneMunya feedback received and applied. Thanks.

MuhanguziDavid commented on package.json:
```diff
@@ -4,11 +4,13 @@
   "scripts": {
-    "start": "nodemon --exec babel-node src/index.js",
+    "build": "babel src --out-dir dist",
+    "start": "node dist/index.js",
```
Looks like your start script will not start the app locally without a user explicitly running the build script. Have your:
- `start` script run the build script then also start the app using nodemon instead of just node (so as to refresh the app when changes are made)
e.g.
`npm run build && nodemon dist/index.js`

MuhanguziDavid commented:
Ensure that social auth works on the deployed app. @Moise1 @T2Wil

MuhanguziDavid commented:
Login and social auth are not working on staging. Do check the links and routes for errors.
